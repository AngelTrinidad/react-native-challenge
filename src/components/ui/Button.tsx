import React from "react";
import {
  StyleSheet,
  ViewStyle,
  TextStyle,
  TouchableOpacity,
} from "react-native";
import { useThemeContext } from "../../contexts/ThemeContext";
import Text from "./Text";

type Props = {
  title: string;
  disabled?: boolean;
  onPress?: () => void;
  style?: ViewStyle;
  textStyle?: TextStyle;
};

const Button: React.FC<Props> = ({
  title,
  onPress,
  style,
  textStyle,
  disabled,
}) => {
  const { colors } = useThemeContext();
  return (
    <TouchableOpacity
      disabled={disabled}
      style={[styles.container, { backgroundColor: colors.primary }, style]}
      onPress={onPress}
    >
      <Text style={[styles.text, textStyle]}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    width: "100%",
    maxWidth: 344,
    minHeight: 42,
    paddingVertical: 8,
    paddingHorizontal: 16,
    borderRadius: 6,
    justifyContent: "center",
    alignItems: "center",
  },
  text: {
    fontSize: 16,
    fontWeight: "500",
    color: "white",
  },
});

export default React.memo(Button);
