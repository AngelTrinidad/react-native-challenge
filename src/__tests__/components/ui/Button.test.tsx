import React from "react";
import renderer from "react-test-renderer";
import Button from "../../../components/ui/Button";

describe("Test Button component", () => {
  it("should render Button", () => {
    const { toJSON } = renderer.create(
      <Button title="Test button" onPress={jest.fn} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
