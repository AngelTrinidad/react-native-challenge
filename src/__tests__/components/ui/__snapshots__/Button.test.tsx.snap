// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component should render Button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1FC4DB",
      "borderRadius": 6,
      "justifyContent": "center",
      "maxWidth": 344,
      "minHeight": 42,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#0A132C",
          "fontSize": 14,
          "fontWeight": "400",
        },
        Object {
          "textAlign": undefined,
        },
        Array [
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "500",
          },
          undefined,
        ],
      ]
    }
  >
    Test button
  </Text>
</View>
`;
